### DISCUSSION TITLE
Use `post-render` only when `quarto render` is used

### DISCUSSION THREAD

**Discussion by skwde** [quarto-dev/quarto-cli#10310] [Q&A] [answered]
### Description

Currently there is no way (at least to my basic knowledge) to run `post-render` only when `quarto render` is used.

Consider following case:
When explicitly calling `quarto render revealjs-slides.qmd` I want to run a `post-render` step calling `decktape` to convert the slides to pdf.
In particular it should not run when I use `quarto preview revealjs-slides.qmd`.

The discussion https://github.com/quarto-dev/quarto-cli/discussions/6080 mentions the use of `QUARTO_PROJECT_RENDER_ALL` but this does not work if there are other `.qmd` files in the project.

Here is a working example.

We have the `_quarto.yml`

```yml
project:
  post-render:
    - test.sh
```

and the `test.sh`

```sh
#!/usr/bin/env bash

echo "Variables:"
echo ${QUARTO_PROJECT_RENDER_ALL}
echo ${QUARTO_PROJECT_OUTPUT_DIR}
echo ${QUARTO_PROJECT_INPUT_FILES}
echo ${QUARTO_PROJECT_OUTPUT_FILES}
```

Following `test.qmd` is used

````qmd
---
format: revealjs
---

# Test

abc
````

In addition there is an empty `another.qmd`.
Running both `render` and `preview` on `test.qmd` shows that the `post-render` is executed

```sh
quarto render test.qmd
quarto preview test.qmd
```

Maybe another variable `QUARTO_PREVIEW`  would be helpful here.

I use Quarto version 1.5.54 in a devcontainer running debian bookworm.

**Comment by mcanouil** [comment, marked as answer]
Thanks!

- https://github.com/quarto-dev/quarto-cli/issues/10314
